herbert_core/utilities/classes/@object_lookup/object_repmat.m

```diff
@@ -0,0 +1,177 @@
+%               input to the Matlab intrinsic function repmat i.e.               
+%               - Valid array size from the Matlab function called size i.e. row
+%                 vectors of at least two integers all of which must be greater
+%                 than or equal to zero.
+%               - A single positive integer n greater than or equal to zero.
+%                 This is equivalent to size vector [n,n].
+%
+%               NOTE:
+%               -----
+%               In the case when there is only one object_array in the
+%               object_lookup and sz_repmat is a cell array {sz1,sz2,...szN},
+%               the number of stored arrays is increased from 1 to N, with the
+%                   1st: repmat(objArr, sz1)
+%                   2nd: repmat(objArr, sz2)
+%                           :
+%                   Nth: repmat(objArr, szN)
+%
+%               This provides a mechanism to create an object_lookup with many
+%               identical stored arrays without the overhead of potentially
+%               expensive multiple checks of object equivalence
+%
+%               EXAMPLE (assuming obj contains just one object array)
+%                   object_repmat (obj, '-repeat', {1,1,1,1,1})
+%
+%
+% Output:
+% -------
+%   obj_out     Output obj_lookup object
+
+
+% Check validity of object_array
+% ------------------------------
+if ~isscalar(obj)
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Only operates on a single object_lookup (i.e. object must be scalar');
+end
+if ~obj.filled
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'The object_lookup is not initialised')
+end
+
+% Parse input
+% -----------
+% Determine number and type OK
+narg = numel(varargin);
+if ~(narg==2 || narg==3) || ~ischar(varargin{end-1})
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Invalid number or type of input argument(s)')
+end
+
+% Get iarray
+if narg==3
+    if ~is_integer_id (varargin{1}) || max(varargin{1})>obj.narray
+        error('HERBERT:object_lookup:invalid_argument', ...
+            'Index array must be contain unique integers in the range 1-%d', obj.narray)
+    end
+    iarray = varargin{1}(:);    % column vector
+else
+    iarray = (1:obj.narray)';   % column vector
+end
+
+% Get sz_repmat
+keyword = varargin{end-1};
+if numel(keyword)<2 || ~strncmpi(keyword, '-repeat', numel(keyword))
+    error('HERBERT:object_lookup:invalid_argument', ...
+        'Unrecognised keyword option %s', keyword)
+end
+sz_repmat = parse_sz_repmat (varargin{end}, numel(iarray), obj.narray); % col vector
+
+
+% Expand arrays
+% -------------
+% Get indexing arrays and sizes of stored arrays
+sz = obj.sz_;
+indx = obj.indx_;
+
+% Reshape the index arrays to the sizes of the object arrays to which repmat
+% will be applied
+sz_sel = sz(iarray);
+indx_sel = cellfun (@(x,y)reshape(x,y), indx(iarray), sz_sel, 'uniformoutput', false);
```

Comment:
Don't need anonymous function for calling with same args
```suggestion
indx_sel = cellfun (@reshape, indx(iarray), sz_sel, 'uniformoutput', false);
```

Reply:
I didn't know that.  It makes code tidier. Now adopted.